### PR TITLE
U-865c3euqh_Configure-LintingPrettier-and-Husky

### PR DESCRIPTION
**Pull request by JoHena**
# Fixes ticket [#865c3euqh](https://app.clickup.com/t/865c3euqh)

## Problem Description
Integrate husky with pre-commit hook prettier and EsLint into the project in order to check code inconsistencies and build

## Checklist 
- [x] Passes eslint linter 
- [x] Passes backend integration tests 
- [x] Added tests for this change 
- [ ] Added documentation for this change 
- [x] Removed unused packages  

## Changes proposed in this pull request
        -Add Husky and Prettier
        -Configure Eslint and Prettier
	-Format existing code

## Reviewers
- @aviada-fgp
- @jorgelabs 
- @aviada-rob 

## Additional Comments
I am thinking of always running the format script before a commit instead of checking if the formatting is correct.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo '🤖 Styling, testing and building your project before committing'
+
+# Check Prettier standards
+npm run check-format ||
+(
+    echo '🤢 Your styling doesnt look soo good.
```
Comment (jorgelabs) [inline .husky/pre-commit:9]: Consider storing error messages in variables for easy editing and reading. This way, if you need to change a message, you only have to do it in one place.
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

echo '🤖 Styling, testing and building your project before committing'

prettierError='🤢 Your styling doesnt look so good. Prettier Check Failed. Run npm run format, add changes and try commit again.'
eslintError='😤 ESLint Check Failed. Make the required changes listed above, add changes and try to commit again.'
typeCheckError='🤡 Failed Type check. Make the changes required above.'
buildError='❌🔨❌ Looks like your build failed ❌🔨❌ Next build failed: View the errors above to see why.'

# Check Prettier standards
npm run check-format || (
    echo "$prettierError"
    false;
)

# Check ESLint Standards
npm run lint || (
    echo "$eslintError"
    false; 
)

# Check tsconfig standards
npm run check-types || (
    echo "$typeCheckError"
    false;
)

# If everything passes... we try to build
echo '🤖 ...Code looks good to me... Trying to build now. 🏗️'

npm run build || (
    echo "$buildError"
    false;
)

# If everything passes... Now we can commit
echo '✅✅✅✅ Everything looks good! I am committing this now. ✅✅✅✅'

```

Comment (JoHena) [reply .husky/pre-commit]: @jorgelabs Added the suggested changes.

**Review by jorgelabs — APPROVED** [review]

**Comment by aviada-fgp** [comment]
Instead of just run the checks, you can run the fixes, for example for the js linter you can run it with --fix param to fix the code while checking, this will help the team to run it before commit the code

**Comment by JoHena** [comment]
> Instead of just run the checks, you can run the fixes, for example for the js linter you can run it with --fix param to fix the code while checking, this will help the team to run it before commit the code

Made the fixes run automatically.

**Review by raulynzunza — APPROVED** [review]
Looks good to me 👍

**Pull request merged**